src/components/CategoryPage.js

```diff
@@ -1,15 +1,23 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { EventsList } from './EventsList';
 
-export const CategoryPage = (props) => (
-  <div>
-    <h2>{props.chosenCategory.name} Events near Toronto, Ontario</h2>
-    <button className="fetch-events" onClick={ () => props.getEvents() }>
-      Fetch Events
-    </button>
-    {props.events 
-      ? <EventsList events={props.events} />
-      : null
-    }
-  </div>
-); 
+export class CategoryPage extends Component {
+  componentDidMount() {
+    this.props.getEvents();
+  }
+
+  render() {
+    return (
+      <div className="category-page">
+        {this.props.allCategories.map(category => category.id === this.props.params.categoryID 
+          ? <h2>{category.name} Events near Toronto, Ontario</h2> 
+          : null)
+        }
+        {this.props.events 
+          ? <EventsList events={this.props.events} />
+          : null
+        }
```

Comment:
If not, we should return an error message or something saying there is no result